docs/src/proposals/standalone-tpu-node.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Standalone TPU node
+---
+
+## Problem
+
+A Solana validator deployment processes all peer-to-peer traffic in the same process.
+This limits the available bandwidth of a validator to that of a single server.
+
+Network interfaces and upstream routers are forced to drop packets when the bandwidth of a validator is exhausted.
+This has been observed during large surges in traffic to the TPU port.
+
+Generic networking setups treat all Solana traffic equal and will randomly drop
+both consensus-relevant traffic (TVU, TPUvote) and TPU traffic.
+The resulting drops in TPUvote and TVU packets result in cluster instability through increased forking.
+
+*Network Diagram: Validator Route*
+
+```
+        +---------+         +--------+        +-----------+
+        |         |         |        |  (!!)  |           |
+--------| Routers |---------| Switch |--------| Validator |
+200G RX |         | 40G RX  |        | 1G RX  |           |
+        +---------+         +--------+        +-----------+
+```
+
+Deploying QoS settings at upstream routers to prioritize consensus-relevant traffic
+is only an option for validator operators that have control of datacenter network infrastructure.
+
+## Proposed Solution
+
+### TPU node
+
+TPU traffic should be processed in a way that cannot affect other validator services.
+
+An effective way to do so is to accept TPU on another physical host than the validator itself
+which is trivially done by advertising a different IP address for the TPU service on the gossip protocol.
+TPU nodes periodically call out to the validator request gossip registration.
+
+*Network Diagram: TPU node route*
+
+```
+        +---------+         +--------+        +-----------+
+        |         |         |        |        |           |
+--------| Routers |---------| Switch |--------| Validator |
+200G RX |         | 40G RX  |        | 1G RX  |           |
+        +---------+         +--------+        +-----------+
+                                 |
+                                 |
+                                 |            +-----------+
+                                 |      (!!)  |           |
+                                 +------------| TPU node  |
+                                       1G RX  |           |
+                                              +-----------+
+```
+
+### TPU proxy protocol
+
+The TPU node needs to forward incoming transactions back to the validator for banking.
+
+It is vital that the TPU node only forwards as many transactions as the validator can process.
+We reintroduce the _SigVerify_ stage to allow the TPU node to drop duplicate and invalid/unsanitized transactions.
+This also offloads signature verification from the validator.
+
+These design constraints require a new _TPUProxy_ protocol with inbuilt congestion control and authentication.
+It exposes a privileged API that allows registration of TPU endpoints and direct submission of transactions to the banking stage.
```

Comment:
The validator will still have to do dedup before letting transactions hit the banking stage.
Each TPU node will only dedup its own stream of transactions but there is no guarantee that multiple connected TPU nodes won't send the same tx multiple times.
Allowing the same tx to hit the banking stage would be a consensus violation.